File: Proline-ORM/src/main/java/fr/proline/core/orm/provider/msi/ORMPeptideInstanceProvider.java
package fr.proline.core.orm.provider.msi;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

import javax.persistence.EntityManager;

import scala.Option;
import scala.collection.Seq;

import fr.proline.core.om.model.msi.PeptideInstance;
import fr.proline.core.om.provider.msi.IPeptideInstanceProvider;
import fr.proline.core.om.provider.msi.IPeptideInstanceProvider$class;
import fr.proline.core.orm.ps.repository.PeptideRepository;

public class ORMPeptideInstanceProvider implements IPeptideInstanceProvider {
	
	private static Map<EntityManager, ORMPeptideInstanceProvider> instances = new HashMap<EntityManager, ORMPeptideInstanceProvider>();
	private PeptideRepository pepRepo = null;
	
	private ORMPeptideInstanceProvider(EntityManager em){
		pepRepo = new PeptideRepository(em);
	}
	
	public static ORMPeptideInstanceProvider getInstance(EntityManager em){
		if(!instances.containsKey(em)){
			instances.put(em, new ORMPeptideInstanceProvider(em));
		}
		return instances.get(em);
	}
	
	@Override
	public Option<PeptideInstance> getPeptideInstance(int pepInstId) {
		return IPeptideInstanceProvider$class.getPeptideInstance(this, pepInstId);				
	}
	
	@Override
	public Option<PeptideInstance>[] getPeptideInstances(Seq<Object> pepInstIds) {
		ArrayList<Option<PeptideInstance>> foundPepInst = new ArrayList<Option<PeptideInstance>>();
		for(int index=0; index<pepInstIds.length();index++){
//			fr.proline.core.orm.ps.
		}
		return null;
	}
	@Override
	public Option<PeptideInstance>[] getResultSummariesPeptideInstances(
			Seq<Object> arg0) {
		// TODO Auto-generated method stub
		return null;
	}
	@Override
	public Option<PeptideInstance>[] getResultSummaryPeptideInstances(int arg0) {
		// TODO Auto-generated method stub
		return null;
	}


}

File: Proline-DAL/src/main/scala/fr/proline/core/om/provider/msi/impl/ORMPeptideInstanceProvider.scala
package fr.proline.core.om.provider.msi.impl

import fr.proline.context.MsiDbConnectionContext
import fr.proline.core.om.model.msi.PeptideInstance
import fr.proline.core.om.provider.msi.IPeptideInstanceProvider
import fr.proline.core.om.util.PeptidesOMConverterUtil

import scala.collection.Seq

class ORMPeptideInstanceProvider( val msiDbCtx: MsiDbConnectionContext) extends IPeptideInstanceProvider {

  val converter: PeptidesOMConverterUtil = new PeptidesOMConverterUtil()

  def getPeptideInstancesAsOptions(pepInstIds: Seq[Long]): Array[Option[PeptideInstance]] = {

    var foundPepInstBuilder = Array.newBuilder[Option[PeptideInstance]]
    pepInstIds foreach (id => {
      val pepInstance: fr.proline.core.orm.msi.PeptideInstance = msiDbCtx.getEntityManager.find(classOf[fr.proline.core.orm.msi.PeptideInstance], id)
      if (pepInstance != null) {
        foundPepInstBuilder += Some(converter.convertPeptideInstanceORM2OM(pepInstance, true, msiDbCtx.getEntityManager))
      } else
        foundPepInstBuilder += None
    })
    return foundPepInstBuilder.result()
  }

  def getPeptideInstances(pepInstIds: Seq[Long] ): Array[PeptideInstance] = {
    this.getPeptideInstancesAsOptions(pepInstIds).filter(_.isDefined).map(_.get)
  }

  def getResultSummariesPeptideInstances(resultSummaryIds: Seq[Long] ): Array[PeptideInstance] = { throw new Exception("NYI") }

}